rescript/get_gtdb.py

```diff
@@ -0,0 +1,138 @@
+    base_tax_url = ('https://data.gtdb.ecogenomic.org/releases/release%s/'
+                    '%s.0/%s_taxonomy_r%s.tsv.gz')
+
+    for version, dcp in ver_dom_dict.items():
+        for dom, cp in dcp.items():
+
+            queries['Taxonomy'].append(
+                (dom,
+                 base_tax_url % (version, version, cp, version),
+                 'FeatureData[Taxonomy]',
+                 'HeaderlessTSVTaxonomyFormat'))
+
+            queries['Sequence'].append(
+               (dom,
+                base_seq_url % (version, version, cp, version),
+                'FeatureData[Sequence]',
+                'DNAFASTAFormat'))
+    return queries
+
+
+def _retrieve_data_from_gtdb(queries):
+    '''
+    Download data from gtdb, given a list of queries.
+
+    queries: {'Taxonomy':(domain, url, type, format), (...),
+              'Sequence':(domain, url, type, format), (...)}
+    '''
+
+    tax_results = []
+    seq_results = []
+
+    print('\nDownloading and processing raw files ... \n')
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        for sttype, q_info in queries.items():
+            for domain, url, dtype, fmt in q_info:
+                print('Retrieving {0} for {1} from {2}'.format(
+                      sttype, domain, url))
+                # grab url
+                bn = os.path.basename(url)
+                destination = os.path.join(tmpdirname, bn)
+                try:
+                    urlretrieve(url, destination)
+                except HTTPError:
+                    msg = ("Unable to retrieve the followng file from GTDB:\n "
+                           + url)
+                    warnings.warn(msg, UserWarning)
+                # seq files are contained within `tar.gz``
+                if tarfile.is_tarfile(destination):
+                    try:
+                        untarred_fn = bn.split('.')[0]+'.fna'
+                        print('  Untarring {0}...\n'.format(bn))
+                        with tarfile.open(destination, 'r') as tar:
+                            tar.extract(member=untarred_fn,
+                                        path=tmpdirname)
+                    except OSError:
+                        pass
+                    seq_results.append(qiime2.Artifact.import_data(
+                                        dtype,
+                                        os.path.join(tmpdirname, untarred_fn),
+                                        fmt))
+                # taxonomy files are contained within ``.gz`
+                else:
+                    try:
+                        unzipped_destination = os.path.splitext(destination)[0]
+                        print('  Unzipping {0}...\n'.format(bn))
+                        _gzip_decompress(destination, unzipped_destination)
+                        destination = unzipped_destination
+                    except OSError:
+                        pass
+                    tax_results.append(qiime2.Artifact.import_data(
```

Comment:
I think those two branches could become two little, independent functions which you could then easily write tests for.